pandas/core/generic.py

```diff
@@ -913,6 +908,13 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
         -------
         renamed : type of caller or None if inplace=True
 
+        Notes
+        -----
+        Prior to version 0.21.0, ``rename_axis`` could also be used to change
+        the axis *labels* by passing a mapping or scalar. This behavior is
+        deprecated and will be removed in a future version. Use ``rename``
+        instead.
+
         See Also
         --------
         pandas.NDFrame.rename
```

Comment:
sidenote: this is useless (does not make a link). Unless you want to start with shared docstring/substitution, I would just explicitly add both for series/frame: `pandas.Series.rename, pandas.DataFrame.rename`